app/hocs/withTranslation.tsx

```diff
@@ -0,0 +1,17 @@
+import type { ComponentType } from 'react'
+import type { Language } from '../hooks/useLanguage'
+import { useLanguage } from '../hooks/useLanguage'
+
+export type EventualString = string | Promise<string> | ((...args: any[]) => (string | Promise<string>));
```

Comment:
I'm unsure about allowing promises here, are there any concrete use cases where that would be nice to have?

Reply:
I thought about it, no there doesn't seem to be any real use cases since the translations are mostly used directly in TSX.